crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -2918,4 +2978,24 @@ mod tests {
 
         assert_is_system(incompatible_system);
     }
+
+    #[test]
+    fn get_components() {
+        #[derive(Component, PartialEq, Eq, Debug)]
+        struct X(usize);
+
+        #[derive(Component, PartialEq, Eq, Debug)]
+        struct Y(usize);
+        let mut world = World::default();
+        let e1 = world.spawn((X(7), Y(10))).id();
+        let e2 = world.spawn(X(8)).id();
+        let e3 = world.spawn_empty().id();
+
+        assert_eq!(
+            Some((&X(7), &Y(10))),
+            world.entity(e1).get_components::<(&X, &Y)>()
+        );
+        assert_eq!(None, world.entity(e2).get_components::<(&X, &Y)>());
+        assert_eq!(None, world.entity(e3).get_components::<(&X, &Y)>());
+    }
```

Comment:
can you add a test to make sure it returns none or panics if you do `world.entity_mut(e).get_components_mut::<&mut X, &mut X>()`?

Reply:
also should have one for `(&mut X, &X)`

Reply:
Wouldn't creating an empty access and calling `update_component_access(state, &mut access)` on it work?

Reply:
It would. It would just be dissatisfying from a perf perspective because Access allocates.

Reply:
FYI I have a PR that removes the allocation from an empty `Access`: https://github.com/bevyengine/bevy/pull/14026/files which might unblock this?

Reply:
Could the allocation not just be removed in the meantime by using a smallvec that has 8 items or something?